lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -438,28 +438,30 @@ def cmd_search(*args)
               cached = true
             end
 
+            if sort_attribute && !valid_sort_attributes.include?(sort_attribute)
+              print_error("Supported options for the -s flag are: #{valid_sort_attributes}")
+              return false
+            end
+
             begin
               if cached
                 print_status('Displaying cached results')
               else
                 search_params = Msf::Modules::Metadata::Search.parse_search_string(match)
                 @module_search_results = Msf::Modules::Metadata::Cache.instance.find(search_params)
 
-                if sort and sort_options.include?(sort)
-                  if sort == 'date'
-                    sort = 'disclosure_date'
-                  end
-                  if sort != 'check'
-                    @module_search_results.sort_by! { |meta| meta.send(sort) }
+                @module_search_results.sort_by! do |module_metadata|
+                  if sort_attribute == 'check'
+                    module_metadata.check ? 0 : 1
+                  elsif sort_attribute == 'disclosure_date' || sort_attribute == 'date'
+                    # Not all modules have disclosure_date, i.e. multi/handler
+                    module_metadata.disclosure_date || Time.utc(0)
```

Comment:
If `module_metadata` doesn't have the `check` attribute or the `disclosure_date` attribute, won't this cause issues here? Should we not be using the safe navigation operator as a result to return `nil` if the field isn't defined, which will work for the case of the comparisons we are doing here?

Reply:
`module_metadata` won't be `nil`, and both the `check` and `disclosure_date` attributes are always present

https://github.com/rapid7/metasploit-framework/blob/619661a845fdd6845f4fae316bf93e9389ff12ab/lib/msf/core/modules/metadata/obj.rb#L47-L48

https://github.com/rapid7/metasploit-framework/blob/619661a845fdd6845f4fae316bf93e9389ff12ab/lib/msf/core/modules/metadata/obj.rb#L19-L20